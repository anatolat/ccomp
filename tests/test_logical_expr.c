int t(int a) {
	printf("%d ", a);
	return a;
}


void tt(int a, int b) {
	int r = t(a) && t(b);
	printf("\na && b = %d\n", r);
	r = t(a) || t(b);
	printf("\na || b = %d\n", r);
}

int main() {
	tt(0, 0);
	tt(0, 1);
	tt(1, 0);
	tt(1, 1);


	printf("\n\n");
	t(1) && t(1) || t(0) && t(0); printf("\n");
	t(0) && t(0) || t(1) && t(1); printf("\n");
}