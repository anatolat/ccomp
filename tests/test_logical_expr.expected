0 
a && b = 0
0 0 
a || b = 0
0 
a && b = 0
0 1 
a || b = 1
1 0 
a && b = 0
1 
a || b = 1
1 1 
a && b = 1
1 
a || b = 1


1 1 
0 1 1 
